receiver/snmpreceiver/client.go

```diff
@@ -0,0 +1,539 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package snmpreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/snmpreceiver"
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/zap"
+
+	"github.com/gosnmp/gosnmp"
+)
+
+// Custom errors
+var (
+	errNoGetOIDs        = errors.New(`all GET OIDs requests have failed`)
+	errNoProcessGetOIDs = errors.New(`all attempts to process GET OIDs have failed`)
+	errNoWalkOIDs       = errors.New(`all WALK OIDs requests have failed`)
+)
+
+// Trimmed down list of SNMP data types to make things a little more simple for the scraper
```

Comment:
nit: I think I'm missing context for this comment.

Reply:
@cpheps i can add some extra context to the comment if it's confusing

Reply:
changed